### PR TITLE
UNet Segmentation

### PR DESCRIPTION
**Pull request by luke-harrison-personal**
Full description of functionality is within README.md

To complete task 1 of the report I implemented the Improved UNet to segment the 2017 ISIC dataset. Unfortunately I wasn't able to train the model as running train.py didn't yield any learning. Running predict.py shows that initial segmentation reflect the shape of the image well, but it scores consistently low when using DICE loss/similarity.

I had consistent issues with memory overflowing which is why .detach() appears often inside modules.py.

### CODE REVIEW

**Comment by luke-harrison-personal** [comment]
I'm cutting it close to the submission time and I'm not expecting feedback since markers will be busy with a large number of submissions so I'm commenting here just so I can potentially be eligible for the 1 point for having comments in the pull request.

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
- Code: mostly OK but model's output activation seems a bit off (leaky_relu->softmax is not very common).
- Results: 0.58 is a bit low and the visualisation is a bit off.
- Readme: OK 
- Commit Messages: OK, most submissions were done in 2 days
- Other comments: N/A

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation (activation function is not optimal) -2
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (poor performance) -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage (issues with visualisation, no plot) -2
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description OK, could be more informative -1
